File: Day 1/python/level 2/LevelTwoTest.py
from level_two import *


def test_get_repeated_values():

    array = [1, 2, 3, 2, 4, 3]

    expected = [2, 3]

    actual = get_repeated_values(array)

    assert actual == expected


def test_get_repeated_values_and_indexes():

    array = [-11, -9, 3, -9, 2, -11]

    expected = [[-11, 0, 5], [-9, 1, 3]]

    actual = get_repeated_values_and_indexes(array)

    assert actual == expected


def test_move_zeros_to_end():

    array = [5, 0, 3, 0, 2, 0]

    expected = [5, 3, 2, 0, 0, 0]

    actual = move_zeros_to_end(array)

    assert actual == expected

File: Day 1/java/level 2/LevelTwoTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

	public class LevelTwoTest {

		@Test
		public void testThatFunctionReturnsRepeatedValues(){

			int[] array = {1,2,3,2,4,3};

			int[] expected = {2,3};
	
			int[] actual = repeatedValues.LevelTwo(array);

			assertArrayEquals(expected, actual);

}

		@Test
		public void testThatFunctionReturnsRepeatedValuesAndIndexes(){

    		int[] array = {-11, -9, 3, -9, 2, -11};

    		int[][] expected = {{-11, 0, 5},{-9, 1, 3}};
        			
    		int[][] actual = LevelTwo.getRepeatedValuesAndIndexes(array);

    		assertArrayEquals(expected, actual);
}

		@Test
		public void testThatFunctionMovesAllZerosToTheEnd(){

    		int[] array = {5, 0, 3, 0, 2, 0};

    		int[] expected = {5, 3, 2, 0, 0, 0};

    		int[] actual = LevelTwo.moveZerosToEnd(array);

    		assertArrayEquals(expected, actual);
}


		












































}

